### PR TITLE
Changes in Dockerfile

### LINKED ISSUE #419: Dockerfile is failing because of insufficient libraries
**Scenario/Intent**
Dockerfile is failing because of insufficient libraries

**Expected Result**
We should be able to build docker image successfully followed by running examples recipes in directory.

### PR DESCRIPTION
**Pull request by chebroluharika**
**Description**
Fixes in Dockerfile

**Issues Resolved**
#419 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [X] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,54 @@
...
+    apt-get install -y --no-install-recommends --no-upgrade && \
+    apt-get install -y ca-certificates curl git vim bash openssl && \
+    apt-get install -y ruby-dev && \
+    apt-get install -y gcc make && \
+    curl -L --progress-bar https://www.chef.io/chef/install.sh | bash -s -- -v 13.12.3
+
+# Some optional but recommended packages
+RUN DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y --no-install-recommends --no-upgrade \
+    # apt-get autoremove -y \
```
Comment (soodpr) [inline Dockerfile]: remove this comment if not required

File: Dockerfile [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
+    git \
+    vim \
+    unzip
+
+RUN gem install oneview-sdk # Ignore the warning about the path not containing gem executables
+
+RUN mkdir -p /chef-repo/.chef/
+RUN mkdir -p /chef-repo/cookbooks
+WORKDIR /chef-repo
+RUN touch knife.rb
```
Comment (soodpr) [inline Dockerfile]: i don't think this line is required.

File: Dockerfile (hunk 1 above)
Comment (chebroluharika) [reply Dockerfile]: done

File: Dockerfile (hunk 2 above)
Comment (chebroluharika) [reply Dockerfile]: done

File: Dockerfile [hunk 3]
```diff
@@ -1,34 +1,52 @@
...
+
+RUN mkdir -p /chef-repo/.chef/
+RUN mkdir -p /chef-repo/cookbooks
+WORKDIR /chef-repo
 RUN echo 'cookbook_path ["#{File.dirname(__FILE__)}/../cookbooks"]' > .chef/knife.rb
 WORKDIR /chef-repo/cookbooks/
-RUN knife cookbook site download compat_resource
+RUN knife cookbook site download compat_resource --force
 RUN tar -xzf compat_resource-*.tar.gz && rm compat_resource*.tar.gz
+RUN git clone https://github.com/HewlettPackard/oneview-chef.git oneview
```
Comment (soodpr) [inline Dockerfile]: instead of git clone command, better to use this command: "ADD . oneview/"

**Review by soodpr — APPROVED** [review]
LGTM 👍  Just gave one more suggestion as we don't need to clone the repo explicitly since we can use ADD current dicrectory. 

Also, can you please add this as an issue in CHANGELOG.md and Open an issue in this repository to track this PR.

**Pull request merged**